qiskit_machine_learning/neural_networks/sampling_neural_network.py

```diff
@@ -26,23 +27,24 @@ class SamplingNeuralNetwork(NeuralNetwork):
     machine learning module that generate samples instead of (expected) values.
     """
 
-    def __init__(self, num_inputs: int, num_weights: int, output_shape: Union[int, Tuple[int, ...]],
-                 return_samples: bool = False) -> None:
+    def __init__(self, num_inputs: int, num_weights: int, dense: bool, return_samples: bool,
+                 output_shape: Union[int, Tuple[int, ...]]) -> None:
         """
 
         Args:
             num_inputs: The number of input features.
             num_weights: The number of trainable weights.
-            output_shape: The shape of the output.
-            return_samples: Determines whether the network returns a batch of samples or a sparse
-                vector (dictionary) of probabilities in its forward pass. In case of probabilities,
+            dense: Returns whether the output is dense or not.
```

Comment:
Similar comment to dense above for the base class; and maybe likewise for return_samples.

Reply:
We have only one example so far, let's not fix a default here but in the implementation.